### PR TITLE
Profile Admin updates

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
CRUD groups and fields + align the UI with https://www.figma.com/file/oo9B9YSe8aWkuLKPBLc0g6/Profile-Generalization?node-id=29%3A2153

Some additional tasks still to tackle in this PR:
- [x] show the profile fields that not as part of a group, and allow a profile to move out of a group completely _(double check if this should be a feature?)_
- [x] show errors correctly from modal
- [x] redirect from the modal
- [x] the order should not change when you save
- [x] when we delete the group, the profile fields move to the Miscellaneous section 
- [x] tests for the groups

Some tasks still to tackle in a subsequent PR:
- [ ] "763vbvgh" - toggle doesn't work when the label is this
- [ ] responsiveness

## Related Tickets & Documents
Part of #6994

## QA Instructions, Screenshots, Recordings

1. Go to http://localhost:3000/admin/profile_fields
2. CRUD the groups and profile fields to test if they do what they supposed to.

screencapture-localhost-3000-admin-profile-fields-2020-09-01-18_25_33

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [x] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
No 

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/views/admin/profile_fields/_profile_field_form.html.erb [hunk 1]
```diff
@@ -1,3 +1,14 @@
+<div class="form-group">
+  <%= form.label :group %>
+  <%= select_tag "profile_field[profile_field_group_id]",
+                 options_from_collection_for_select(
+                   ProfileFieldGroup.all,
+                   :id, :name,
+                   selected: !group ? nil : group.id
```
Comment (citizen428) [inline app/views/admin/profile_fields/_profile_field_form.html.erb:7]: **suggestion (non-blocking):** I think `gropu&.id` or `group.try(:id)` would be easier to read than this ternary.

**Review by citizen428 — APPROVED** [review]
Very nice @Ridhwana! I left a couple of small comments/questions, feel free to address them here, make a follow up PR or ignore them as you see fit. I also restarted the build since the failing specs seemed to be unrelated. May the Travis odds be ever in your favor.

File: app/javascript/chat/chat.jsx [hunk 2]
```diff
@@ -130,7 +130,9 @@ export default class Chat extends Component {
 
     setupObserver(this.observerCallback);
 
-    this.subscribePusher(`private-message-notifications--${appName}-${currentUserId}`);
```
[the remark this answers is not in the document]
Comment (Ridhwana) [reply app/javascript/chat/chat.jsx:133]: Done.✅  My linter seems to be over eager

File: app/controllers/admin/profile_fields_controller.rb [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
     ALLOWED_PARAMS = %i[
-      input_type label active placeholder_text description group
+      input_type label active placeholder_text description profile_field_group_id
     ].freeze
     layout "admin"
 
     def index
-      @grouped_profile_fields = ProfileFieldGroup.all.includes(:profile_fields)
+      @grouped_profile_fields = ProfileFieldGroup.all.includes(:profile_fields).order(:created_at)
+      @ungrouped_profile_fields = ProfileField.where(profile_field_group_id: nil).order(:created_at)
```
[the remark this answers is not in the document]
Comment (Ridhwana) [reply app/controllers/admin/profile_fields_controller.rb:10]: Good idea, I'll evaluate this in my next PR.

File: app/views/admin/profile_fields/_profile_field_form.html.erb (hunk 1 above)
Comment (Ridhwana) [reply app/views/admin/profile_fields/_profile_field_form.html.erb:7]: I always forget ruby has this neat trick. Thanks!

File: app/controllers/admin/profile_fields_controller.rb [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
   class ProfileFieldsController < Admin::ApplicationController
     ALLOWED_PARAMS = %i[
-      input_type label active placeholder_text description group
+      input_type label active placeholder_text description profile_field_group_id
     ].freeze
     layout "admin"
 
     def index
-      @grouped_profile_fields = ProfileFieldGroup.all.includes(:profile_fields)
+      @grouped_profile_fields = ProfileFieldGroup.all.includes(:profile_fields).order(:created_at)
```
[the remark this answers is not in the document]
Comment (Ridhwana) [reply app/controllers/admin/profile_fields_controller.rb:9]: I think so, I used name and for the ungrouped ones, i used label.

**Review by rhymes — APPROVED** [review]
Works very well, left a couple of non blocking notes :)

File: app/views/admin/profile_fields/_add_group_modal.html.erb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            <%= form.submit class: "btn btn-primary" %>
+          <% end %>
+        </div>
+      </div>
+    </div>
+    <div class="crayons-modal__overlay"></div>
+  </div>
+</div>
+
+<script>
```
[the remark this answers is not in the document]
Comment (Ridhwana) [reply app/views/admin/profile_fields/_add_group_modal.html.erb:36]: Hmm, I do prefer the js to be split from the erb. I wasn't sure what the discipline was within the codebase. If this is the case, I'll look into it and I can move it. 👍🏽

File: app/controllers/admin/profile_fields_controller.rb [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
   class ProfileFieldsController < Admin::ApplicationController
     ALLOWED_PARAMS = %i[
-      input_type label active placeholder_text description group
+      input_type label active placeholder_text description profile_field_group_id
     ].freeze
     layout "admin"
 
     def index
-      @grouped_profile_fields = ProfileFieldGroup.all.includes(:profile_fields)
+      @grouped_profile_fields = ProfileFieldGroup.all.includes(:profile_fields).order(:name)
```
[the remark this answers is not in the document]
Comment (Ridhwana) [reply app/controllers/admin/profile_fields_controller.rb:9]: Oops I missed this @rhymes. I'll add it to my next PR. Thanks!

**Pull request merged**